dom.go

```diff
@@ -142,9 +191,11 @@ func (h *HTML) restoreHTML(prev *HTML) {
 			h.node.Set(name, value)
 		}
 	}
-	for name := range prev.properties {
-		if _, ok := h.properties[name]; !ok {
-			h.node.Set(name, nil)
+	if !h.new {
+		for name := range prev.properties {
```

Comment:
I wonder if `h.new` is needed at all, actually. Won't `prev` always be `nil` in the case of new node creation? That would make all of our `range` statements here no-op, and then we can save ourselves the trouble of keeping the `h.new` state variable (and we save the memory + defer performance cost).

For readability sake, we can of course add a comment above these all (or above each one, whichever seems more clear) stating this.